internal/compatcontext/cancel_go121.go

```diff
@@ -0,0 +1,10 @@
+//go:build go1.21
+// +build go1.21
```

Comment:
Looks like we need a backward-combatiblity wrapper for older Go versions

```
1.493 package github.com/docker/docker/cmd/dockerd
1.493 	imports github.com/docker/docker/daemon/cluster
1.493 	imports github.com/docker/docker/internal/compatcontext: build constraints exclude all Go files in /go/src/github.com/docker/docker/internal/compatcontext
1.497 + rm -f /go/src/github.com/docker/docker/go.mod
```

Reply:
OMFG That is what this package is for but somehow it's missing a file?

Reply:
Yes, was wondering if you forgot to stage the counterpart of this one

(Initially I was 😱 and thought we made a mistake in the single flight contribution, but I was "happy" to see it was here in our repository 😅)

Reply:
Should be good now.